catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -115,6 +116,15 @@
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
+		if (getIntent().getExtras() != null) {
+			try {
+				String link = getIntent().getExtras().get("link").toString();
+				startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(link)));
+			} catch (NullPointerException e) {
+				Log.e(TAG, "Url not found with notification", e);
+			}
+		}
```

Comment:
Please create a method.
___
There is a problem if the link doesn't start with http:// or https://. The app stays in an infinite loop of creating the MainMenuActivity, crashing and restarting the App.
___
Open any Activity besides the MainMenuActivity, e.g. the ScriptActivity. Move the app into the background. Send yourself a message and tap on the message, the browser opens. Press back. The MainMenuAtivity is shown. Press back again, the ScriptActivity is shown.
The MainMenuActivity shouldn't be shown when pressing back the first time.